packages/shared/src/components/streak/popup/ReadingStreakPopup.tsx

```diff
@@ -9,19 +9,46 @@ import { getReadingStreak30Days, UserStreak } from '../../../graphql/users';
 import { useAuthContext } from '../../../contexts/AuthContext';
 import { Weekends } from '../../../lib/dateFormat';
 
-const today = new Date();
-const dateToday = today.getDate();
-const streakDays = [
-  subDays(today, 4),
-  subDays(today, 3),
-  subDays(today, 2),
-  subDays(today, 1),
-  today,
-  addDays(today, 1),
-  addDays(today, 2),
-  addDays(today, 3),
-  addDays(today, 4),
-]; // these dates will then be compared to the user's post views
+const getStreak = ({
+  isCompleted,
+  isToday,
+  isFreezeDay,
+}: {
+  isCompleted: boolean;
+  isToday: boolean;
+  isFreezeDay: boolean;
+}) => {
+  if (isCompleted) {
+    return Streak.Completed;
+  }
+
+  if (isFreezeDay) {
+    return Streak.Freeze;
+  }
+
+  if (isToday) {
+    return Streak.Pending;
+  }
+
+  return Streak.Upcoming;
+};
```

Comment:
Just a note, might be wrong, but this looks like a additional state transformation that does not do a lot. Maybe getStreak could actually contain [below logic](https://github.com/dailydotdev/apps/pull/3075/files#diff-75b6b950d2d429c84bce12f44b6b3980637e4c95ff04ce893869b73ea551950aR82-R94) and then return `Streak` enum.